pkg/miniogw/gateway-storj.go

```diff
@@ -72,48 +204,106 @@ func (s *storjObjects) GetObject(ctx context.Context, bucket, object string,
 }
 
 func (s *storjObjects) GetObjectInfo(ctx context.Context, bucket,
-	object string) (objInfo cmd.ObjectInfo, err error) {
+	object string) (objInfo minio.ObjectInfo, err error) {
 	panic("TODO")
 }
 
 func (s *storjObjects) ListBuckets(ctx context.Context) (
-	buckets []cmd.BucketInfo, err error) {
-	return []cmd.BucketInfo{{
-		Name:    "test-bucket",
-		Created: time.Now(),
-	}}, nil
+	buckets []minio.BucketInfo, err error) {
+	return s.getBuckets()
 }
 
 func (s *storjObjects) ListObjects(ctx context.Context, bucket, prefix, marker,
-	delimiter string, maxKeys int) (result cmd.ListObjectsInfo, err error) {
-	return cmd.ListObjectsInfo{
-		IsTruncated: false,
-		Objects: []cmd.ObjectInfo{{
-			Bucket:      "test-bucket",
-			Name:        "test-file",
-			ModTime:     time.Now(),
-			Size:        0,
-			IsDir:       false,
-			ContentType: "application/octet-stream",
-		}},
-	}, nil
+	delimiter string, maxKeys int) (result minio.ListObjectsInfo, err error) {
+	return s.getFiles(bucket)
 }
 
 func (s *storjObjects) MakeBucketWithLocation(ctx context.Context,
 	bucket string, location string) error {
 	panic("TODO")
 }
 
+//encryptFile encrypts the uploaded files
+func encryptFile(data io.ReadCloser, blockSize uint, bucket, object string) error {
+	dir := os.TempDir()
+	dir = filepath.Join(dir, "gateway", bucket, object)
+	err := os.MkdirAll(dir, 0755)
+	if err != nil {
+		return err
+	}
+	fc, err := infectious.NewFEC(*rsk, *rsn)
+	if err != nil {
+		return err
+	}
+	es := eestream.NewRSScheme(fc, *pieceBlockSize)
+	encKey := sha256.Sum256([]byte(*key))
+	var firstNonce [12]byte
+	encrypter, err := eestream.NewAESGCMEncrypter(
+		&encKey, &firstNonce, es.DecodedBlockSize())
+	if err != nil {
+		return err
+	}
+	readers, err := eestream.EncodeReader(context.Background(), eestream.TransformReader(
+		eestream.PadReader(os.Stdin, encrypter.InBlockSize()), encrypter, 0),
```

Comment:
Why did you change reading from `data` to `os.Stdin`? This broke the upload functionality.

Reply:
ooo...man... merge overlook...thnx....it happened because I had to merge the latest changes and the function call's parameters were changed and I had to pull in those changes...well hope this is the last change.